ttnn/cpp/ttnn/operations/normalization/softmax/device/softmax_device_operation.hpp

```diff
@@ -0,0 +1,116 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include "common/base_types.hpp"
+#include "common/core_coord.h"
+#include "ttnn/tensor/types.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/operation.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/decorators.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/compute_kernel_config.hpp"
+
+
+#include "softmax_types.hpp"
+
+namespace ttnn::operations::normalization {
+
+struct SoftmaxDeviceOperation {
+    struct operation_attributes_t {
```

Comment:
In my PR I moved these args to a common place and completely moved out programs to their own .hpp and .cpp